src/Metadata/Property/Factory/AnnotationPropertyNameCollectionFactory.php

```diff
@@ -63,7 +63,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
         // Properties
         foreach ($reflectionClass->getProperties() as $reflectionProperty) {
             if (null !== $this->reader->getPropertyAnnotation($reflectionProperty, ApiProperty::class)) {
-                $propertyNames[$reflectionProperty->name] = true;
+                $propertyNames[$reflectionProperty->name] = (string)$reflectionProperty->name;
```

Comment:
I studied this a bit more, looks like the fix should be in `PropertyNameCollection`:

```diff
--- a/src/Metadata/Property/PropertyNameCollection.php
+++ b/src/Metadata/Property/PropertyNameCollection.php
@@ -30,7 +30,9 @@ final class PropertyNameCollection implements \IteratorAggregate, \Countable
      */
     public function __construct(array $properties = [])
     {
-        $this->properties = $properties;
+        $this->properties = array_map(function($property) {
+            return (string) $property;
+        }, $properties);
     }
```